### PR TITLE
OVERHAUL PT. 3: ENERGY ARMORY REMADE

### PR DESCRIPTION
**Pull request by Draganfrukts**
### THIS PR IS PART OF THE ARMORY OVERHAUL PROJECT. SPRITES BY ELECTRICPANTS (@slatrhaus) AND DRAGONFRUITS (@dragonfruits.)
- Re-sprites every relevant energy weapon in the game
- Nukes redundant bloat weapons
- Overhauls certain names and descriptions

### A unique weapon, reflavored.
The **_Solar Scorcher_** has been replaced by **Pew Pew**.
Pew Pew is a unique, powerful laser pistol that deals insane damage at the cost of having HORRIFYING ammo economy.

As suggested by its name, **Pew Pew can only fire two shots before depleting its cell.**
Every time one of Pew Pew's bolts connects with a target, it will deal **45 damage** at **45% armor penetration**.

### Show me pictures.
image
_...And more!_

I also got rid of Sunset's custom loadouts as they don't apply here.

### CODE REVIEW

File: code/modules/projectiles/projectile/beams.dm [hunk 1]
```diff
@@ -390,6 +390,16 @@
 	muzzle_type = /obj/effect/projectile/muzzle/laser
 	impact_type = /obj/effect/projectile/impact/laser
 
+/obj/item/projectile/beam/laser/pistol/hitscan/pewpew
+	name = "overtuned laser beam"
+	damage = 35
```
Comment (Draganfrukts) [inline code/modules/projectiles/projectile/beams.dm:395]:
```suggestion
	damage = 45
```

File: code/modules/projectiles/projectile/beams.dm [hunk 2]
```diff
@@ -390,6 +390,16 @@
...
+	wound_bonus = 15
+	bare_wound_bonus = 25
+	armour_penetration = 0.25
```
Comment (Draganfrukts) [inline code/modules/projectiles/projectile/beams.dm:398]:
```suggestion
	armour_penetration = 0.45
```

File: code/modules/projectiles/guns/energy/plasmaf13.dm [hunk 3]
```diff
@@ -7,10 +7,11 @@
...
 	name ="plasma pistol"
 	slowdown = 0.1
-	armour_penetration = 0.02
 	item_state = "plasma-pistol"
-	icon_state = "plasma-pistol"
-	desc = "A pistol-sized miniaturized plasma caster built by REPCONN. It fires a bolt of superhot ionized gas."
+	fire_delay = 2
+	icon = 'icons/obj/guns/gunfruits2022/energy.dmi'
+	icon_state = "plaspistol"
+	desc = "A REPCONN-made pistol-sized plasma caster based on a Winchester model. Fires bolts of plasma that form into toroids in the chamber, which are then sent down a superconducting barrel and propelled by electromagnetic claws."
```
[the remark this answers is not in the document]
Comment (Draganfrukts) [reply code/modules/projectiles/guns/energy/plasmaf13.dm:14]: Non-issue for now, anyone is free to switch the order around once the PR's done

File: code/modules/projectiles/guns/energy/laser.dm [hunk 4]
```diff
@@ -131,7 +131,7 @@
 
 
 
-//Fallout
+//	v	FALLOUT 13 CONTENT STARTS HERE	v
```
[the remark this answers is not in the document]
Comment (Draganfrukts) [reply code/modules/projectiles/guns/energy/laser.dm:134]: If someone wants to refactor the Fallout gun files to be completely in their own directories, they're free to.

File: code/modules/projectiles/guns/energy/laser.dm [hunk 5]
```diff
@@ -411,9 +340,10 @@
 
 //AER9 Laser rifle
 /obj/item/gun/energy/laser/aer9
-	name = "\improper AER9 laser rifle"
-	desc = "A sturdy pre-war laser rifle. Emits beams of concentrated light to kill targets. Fast firing, but not very powerful."
-	icon_state = "laser"
+	name = "laser rifle"
```
[the remark this answers is not in the document]
Comment (Draganfrukts) [reply code/modules/projectiles/guns/energy/laser.dm:343]: > i actually disagree, if you say 'aer9' everyone knows what it is but if you say 'laser rifle' there's ambiguity

The games after Fallout 2 & Tactics stop mentioning guns by their model names. A few exceptions are reserved for uniques and that's it. I think it's infinitely better for the player

File: code/modules/projectiles/guns/energy/laser.dm [hunk 6]
```diff
@@ -1,4 +1,4 @@
-/obj/item/gun/energy/laser/practice
+/obj/item/gun/energy/laser/practice				//SCROLL DOWN FOR FALLOUT 13 CONTENT
```
[the remark this answers is not in the document]
Comment (Draganfrukts) [reply code/modules/projectiles/guns/energy/laser.dm:1]: IDFK non-issue

**Comment by maaacha** [comment]
@Draganfrukts is there any further changes you intend to add to this PR, or is it effectively done?

**Review by maaacha — APPROVED** [review]

**Pull request merged**